iocore/net/ActionProcessor.cc

```diff
@@ -0,0 +1,54 @@
+/** @file
+
+  A brief file description
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+/*************************** -*- Mod: C++ -*- ******************************
+ ActionProcessor.cc
+   Created On      : 05/02/2017
+
+   Description:
+   SNI based Configuration in ATS
+ ****************************************************************************/
+#include "P_ActionProcessor.h"
+#include "ts/ink_memory.h"
+#include "P_SSLSNI.h"
+#include "P_Net.h"
+#include "P_SSLNextProtocolAccept.h"
+#include "P_SSLUtils.h"
+
+extern Map<int, SSLNextProtocolSet *> snpsMap;
+
+void
+ActionProcessor::PerformAction(Continuation *cont, cchar *servername)
```

Comment:
Unfortunately this name is likely to get bonged, because of class `Action` in `iocore/eventsystem/I_Action.h`.